assembl/static2/js/app/reducers/adminReducer/adminSections.js

```diff
@@ -61,7 +63,30 @@ export const sectionsById = (state: Map<string, Map> = Map(), action: ReduxActio
   case CREATE_SECTION:
     return state.set(action.id, defaultResource.set('id', action.id).set('order', action.order));
   case DELETE_SECTION: {
-    return state.setIn([action.id, 'toDelete'], true);
+    const sections = state.sort((a, b) => {
+      const aOrder = a.get('order');
+      const bOrder = b.get('order');
+      return aOrder - bOrder;
+    });
+    const idToDelete = state.getIn([action.id, 'id']);
+    let newState = Map();
+    let count = -1;
+    sections.forEach((s) => {
+      if (s.get('id') !== idToDelete && s.get('type') !== 'ADMINISTRATION') {
+        count += 1;
+        const sectionInfo = Map({
+          isNew: false,
+          order: count,
+          id: s.get('id'),
+          titleEntries: s.get('titleEntries'),
+          url: s.get('url'),
+          type: s.get('type')
+        });
+        newState = newState.set(s.get('id'), sectionInfo);
+      }
+    });
```

Comment:
really not sure this is a good way to change the order value of all remaining items

Reply:
it changed a lot since, we don't use `order` in `sectionsById` state anymore (everything related to order is done in `sectionsInOrder`). We just reset the order of each section based on index in `sectionsInOrder` before to send the mutations